### PR TITLE
Change clusterConfig API version and family

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR changes the group and version of ClusterConfig API (Ref. #188)

# How Has This Been Tested?

This refactoring has been tested with unit tests and E2E testing pipeline.

### CODE REVIEW

File: api/config/v1alpha1/groupversion_info.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 
 var (
 	// GroupVersion is group version used to register these objects
-	GroupVersion = schema.GroupVersion{Group: "policy.liqo.io", Version: "v1"}
+	GroupVersion = schema.GroupVersion{Group: "config.liqo.io", Version: "v1"}
```
Comment (fraborg) [inline api/config/v1alpha1/groupversion_info.go:28]:
```suggestion
	GroupVersion = schema.GroupVersion{Group: "config.liqo.io", Version: "v1alpha1"}
```

File: deployments/liqo_chart/crds/config.liqo.io_clusterconfigs.yaml [hunk 2]
```diff
@@ -6,9 +6,9 @@ metadata:
   annotations:
```
Comment (fraborg) [inline deployments/liqo_chart/crds/config.liqo.io_clusterconfigs.yaml:6]: I think we need a `make -f scripts/cluster-config/Makefile manifests ` to regenerate the crd, there is still v1

**Review by fraborg — APPROVED** [review]

**Review by AbakusW — APPROVED** [review]

File: api/config/v1alpha1/groupversion_info.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 // Package v1 contains API Schema definitions for the policy v1 API group
 // +kubebuilder:object:generate=true
 // +groupName=policy.liqo.io
```
Comment (fraborg) [inline api/config/v1alpha1/groupversion_info.go:18]:
```suggestion
// +groupName=config.liqo.io
```

**Pull request merged**